test/xbgpu/test_engine.py

```diff
@@ -1596,68 +1595,57 @@ async def test_capture_stop_some_streams(
-            for stopped_corrprod in stopped_corrprods:
-                capture_stop_corrprod_request = ("capture-stop", stopped_corrprod.name)
-                stopped_xpipeline = xbengine._request_pipeline(stopped_corrprod.name)[0]
-                # NOTE: We patch the instance and not the class in this case as we only
-                # want each corrprod stream (XPipeline) to affect its own output.
-                self._patch_method(
-                    monkeypatch,
-                    stopped_xpipeline.send_stream,
-                    "get_free_heap",
-                    # TODO: NGC-1568 to remove the `+ 1` appended here
-                    capture_stop_accum_index + 1,
-                    client,
-                    [capture_stop_corrprod_request],
-                )
-
-        beam_capture_stop_heap_indices: list[int | None] = [None] * len(beam_outputs)
-        if n_b_streams_to_stop is not None:
-            assert n_b_streams_to_stop <= len(beam_outputs)
-            # NOTE This value is arbitrarily chosen, but must be less than
-            # `n_batches` / `HEAPS_PER_FENGINE_PER_CHUNK` to actually take effect.
-            capture_stop_chunk_index = 10
-            assert capture_stop_chunk_index < n_batches // HEAPS_PER_FENGINE_PER_CHUNK
-            if n_b_streams_to_stop == 1:
-                stream_id_to_stop = rng.integers(0, len(beam_outputs))
-                stopped_beams = [beam_outputs[stream_id_to_stop]]
-                beam_capture_stop_heap_indices[stream_id_to_stop] = (
-                    capture_stop_chunk_index * HEAPS_PER_FENGINE_PER_CHUNK
-                )
-            else:
-                stopped_beams = beam_outputs
-                beam_capture_stop_heap_indices = [capture_stop_chunk_index * HEAPS_PER_FENGINE_PER_CHUNK] * len(
-                    beam_outputs
-                )
-            capture_stop_beams = []
-            for stopped_beam in stopped_beams:
-                capture_stop_beams.append(("capture-stop", stopped_beam.name))
-            # NOTE: `get_free_chunk` is given `capture_stop_index` + 1 because
-            # `patch_method` counts from 1 instead of 0.
+        assert n_x_streams_to_stop <= len(corrprod_outputs)
+        # NOTE: This value is arbitrarily chosen, but must be less than
+        # `n_total_accumulations` in order to accurately test stopping of
+        # corrprod data streams.
+        capture_stop_accum_index = 3
+        assert capture_stop_accum_index < n_total_accumulations
+        stream_ids_to_stop = sorted(rng.choice(len(corrprod_outputs), n_x_streams_to_stop, replace=False))
+        stopped_corrprods: list[XOutput] = []
+        for stream_id in stream_ids_to_stop:
+            stopped_corrprods.append(corrprod_outputs[stream_id])
+            corrprod_capture_stop_accum_indices[stream_id] = capture_stop_accum_index
+        for stopped_corrprod in stopped_corrprods:
```

Comment:
I think this can be simplified a little now, since you're building up `stopped_corrprods` just to loop over it immediately after:
```suggestion
        for stream_id in stream_ids_to_stop:
            stopped_corrprod = corrprod_outputs[stream_id]
            corrprod_capture_stop_accum_indices[stream_id] = capture_stop_accum_index
```

Reply:
Ah, too right. My apologies, I was still holding on to the idea that I was using `stopped_corrprods` later on when verifying data. Good spot.